src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs

```diff
@@ -1407,6 +1403,10 @@ void IEnumerator.Reset()
             {
                 _index = -1;
             }
+
+            public void Dispose()
```

Comment:
This is a size optimization. Moving the Dispose method to the generic type means that the method body will be duplicated for each array type. If it's in the non-generic base type, there's only one body. Why is this change needed?

Reply:
It was a bug in the analyzer (reverted the change)